src/utils/fileUtils.ts

```diff
@@ -97,8 +97,8 @@ export function sortCompare(fileInfo1: FileInfo, fileInfo2: FileInfo, key: strin
 	}
 	// finally sort by name
 	return asc
-		? fileInfo1[key].localeCompare(fileInfo2[key], getLanguage(), { numeric: true })
-		: -fileInfo1[key].localeCompare(fileInfo2[key], getLanguage(), { numeric: true })
+		? fileInfo1[key].toString()?.localeCompare(fileInfo2[key].toString(), getLanguage(), { numeric: true })
+		: -fileInfo1[key].toString()?.localeCompare(fileInfo2[key].toString(), getLanguage(), { numeric: true })
```

Comment:
Not sure if `-undefined` makes sense. This is guaranteed to always be a string.

```suggestion
		? String(fileInfo1[key]).localeCompare(fileInfo2[key].toString(), getLanguage(), { numeric: true })
		: -String(fileInfo1[key]).localeCompare(fileInfo2[key].toString(), getLanguage(), { numeric: true })
```

Reply:
From the original PR, it is to match https://github.com/nextcloud/photos/pull/783, which makes sense to keep similar.